### ISSUE TITLE
Uncompressed document-associated bytes

### ISSUE THREAD

**Issue by jason-wolfe** [quickwit-oss/tantivy#267]
Sometimes things don't quite fit into things built in to tantivy and it would be helpful to have uncompressed document-associated bytes that tantivy is aware of.

Example use-cases are:
* things where some specialized compression is preferred
* multi-dimensional floating point values associated with a document needed for filtering (e.g., lat/long or xyz coordinates)
* (1-per-document) precomputed floating point partial score value to aid in scoring

These need to be fast to look up, and tantivy doesn't need to do anything smart with them, aside from propagate them in a merge.

Basically, it's a catch-all case which serves as a strict improvement over storing these in random docid->value files on disk.

**Why not stored values?** Because decompressing whole blocks is too slow for matching/scoring.

**Why does tantivy need to be aware?** For documents to be tracked and valid across merges.

**Comment by fulmicoton** [comment]
I am not sure if you want byte array fast fields : `&[u8]` or family columns in fastfields. Can you detail?

**Comment by jason-wolfe** [comment]
`&[u8]` fast fields are what I'm thinking of here. Just some general mechanism for doing things that are not yet integrated into tantivy or deliberately not implemented in tantivy.

Family columns in fast fields could be useful to mitigate a lot of semi-related locality issues, though!

**Comment by fulmicoton** [comment]
Both are very valid features, but let's focus on `&[u8]` fast fields as it is the more versatile one.

**Comment by fulmicoton** [comment]
Requalified in #270
